src/Resolver.js

```diff
@@ -159,7 +159,8 @@ export default class Resolver extends React.Component {
         const factory = resolve[name];
         const value = factory(props);
 
-        if (value instanceof Promise) {
+        // duckpunched promise
+        if (value instanceof Promise || typeof value.then === 'function') {
```

Comment:
Can you switch to double-quotes to match to match the rest of the coding style?  (I guess I forgot to `.eslintrc` this)